### ISSUE TITLE
pybigwig regression writing bigwigs in >=0.1.1

### ISSUE THREAD

**Issue by sergpolly** [jackh726/bigtools#27]
looks like there was a regression from 0.1.0 to 0.1.1+ when trying to create a bgiwig - using example code from this repo:
```python
import math
import random
import pybigtools

def genintervals(clengths):
    for chrom in clengths:
        clength = clengths[chrom]
        current = random.randint(0, 300)
        start = current
        while True:
            length = random.randint(1, 200)
            end = start + length
            if end > clength:
                break
            value = round(random.random(), 5)
            yield (chrom, start, end, value)
            start = end + random.randint(20, 50)

chroms = ["chr1", "chr2", "chr3"]
clengths = {"chr1": 10000, "chr2": 8000, "chr3": 6000}
intervals = list(genintervals(clengths))

b = pybigtools.open("test-xxx-delete.bw", "w")
b.write(clengths, iter(intervals))
```
this works fine with 0.1.0 and just hangs real bad with 0.1.1+ ( i'm working from the jupyter notebook - and have to restart it, as interrupt doesn't work ) ...
bigbed seems to work fine though across versions ...

**Comment by jackh726** [comment]
okay I think I figured out the problem and fixed it.

Trying to make sure the python tests get run in CI, then I'll release a new version

**Comment by sergpolly** [comment]
> Trying to make sure the python tests get run in CI, then I'll release a new version

Awesome ! Thank you !

it works fine otherwise - definitely a nice replacement for ucsc-tools, which are a pain to install with conda ...

**Comment by nvictus** [comment]
Fixed in v0.1.3. Re-open if there are still issues.
